sql/tests: TestRandomSyntaxSQLSmith failed

sql/tests.TestRandomSyntaxSQLSmith [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/19816571?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/19816571?buildTab=artifacts#/) on release-25.2.1-rc @ [893b4a9651bdf5c05e1884db8956e312f91d7075](https://github.com/cockroachdb/cockroach/commits/893b4a9651bdf5c05e1884db8956e312f91d7075):

Random syntax error:

```
    rsg_test.go:965: Crash detected: server panic: pq: internal error: runtime error: slice bounds out of range [-9223372036854775807:]
```
Query:

```
        WITH
        	with_129817 ("c	ol_766093")
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						('2016-01-18 05:35:05.000347+00':::TIMESTAMPTZ),
        						(
        							('1990-07-21 23:15:05.000637+00':::TIMESTAMPTZ::TIMESTAMPTZ - '23 years 4 mons 400 days 04:36:58.863706':::INTERVAL::INTERVAL)::TIMESTAMPTZ
        						),
        						('1980-04-13 02:25:22.000899+00':::TIMESTAMPTZ),
        						(('19:27:13.607263+06:38':::TIMETZ::TIMETZ + '1970-01-01':::DATE::DATE)::TIMESTAMPTZ),
        						('2020-06-02 04:14:37.000341+00':::TIMESTAMPTZ),
        						(NULL)
        				)
        					AS "tab _319421" ("c	ol_766093")
        		),
        	with_129818 (col_766094)
        		AS (
        			SELECT
        				*
        			FROM
        				(
        					VALUES
        						(e'\x0fKv\x05+':::STRING),
        						(e'q\nu^:,y9':::STRING),
        						(e'w/\x10C()Tk/':::STRING),
        						(current_database()::STRING),
        						(
        							substring_index(('[{"foobar": "Sl6k"}, {"0X4Hlz": {"b": false}, "PIY6Zs": false, "bar": null}, true, 3.0770293825693096, {"bar": null}, "bar", true]':::JSONB::JSONB || '':::STRING::STRING)::STRING::STRING, '``My(Vcs{i':::STRING::STRING, ((-6648654279772771880):::INT8::INT8 & (-9223372036854775807):::INT8::INT8)::INT8::INT8)::STRING
        						)
        				)
        					AS 😯tab_319422 (col_766094)
        		)
        SELECT
        	('\xb8d6':::BYTES, ARRAY['3297c92d-5f2a-4460-bde7-f4a1c0aa9cf5':::UUID,'199d9ed4-95d0-4bac-80af-16af3c37002a':::UUID,'fb8a2867-3f87-4f40-95e8-744f059d5436':::UUID,'ecab8ba9-33c4-4a08-86eb-d40d304c37c8':::UUID,'ffffffff-ffff-ffff-ffff-ffffffffffff':::UUID], e'\x05A\x14':::STRING, 1577493043:::REGROLE, 'xaC':::STRING, 'BOX(-0.16600108774763822 -0.23954838398692058,0.3035878511648994 0.35829919800709287)':::BOX2D, '[-8.917565e-38,2.3367464e-32,-0.0001033443,-1.4927306e+21,-3.2743734e+37,-7.4058886e+30,-1.2601797e-05,4.9619562e-08,1.3837337e+27,-11.01728,1.6229885e+19,-3.778754e+06,5.4528976e+13,7.725168e-05,9.548917e-25,1.1074495e-09,-1.5997938e-18,-1.02268825e-36,3.9656758e-14,-1.4255255e+32,-1.5120375e+34,-3.732117e+34,-1.4551317e-30,-6.586803e-27,2.0936476e-33,1.965748e-29,8.545701e+27,-2.599904e-31,-1.1598359e-26,-3.1584576e+23,-8.63081e-22,3.4212528e+23,-172.87166,1.1897876e+37,2.8360338e-26,-1.317159e+06,1.7175511e+30,-9.696445e+26,1.4444369e+11,-1.2473789e-20,-0.0013925248,7.2973415e+06,1.2835136e+09,2.0738969e+27,-32497.578,2.5083966e-34,7.954503e-37,-7.6555036e-22,3.6181536e+20,-1.0813577e+09,-7.951387e+06,0.00027138653,1.9713225e+17,-7.448095e+34,1.1783547e-21,4.9422997e+30,6.4567997e+35,6.0957827e-30,0.20315501,4.1330952e-22,3.1226004e+33,-1.6496222e+17,9.258849e+27,-4.0069658e-17,-2.3029009e+18,-2.9281471e+38,127133.67,-5.43226e-15,7.855775e+17,-3.908981e+34,9.6856625e-14,-1.22494706e+11,-2.813516e+09,-1.6208904e+35,5.781882e-07,-3.7868146e+27,-8.085691e-05,2.3247908e-08,0.04831349,583.83453,-3.6272668e-10,123.54401,1.2781608e-36,0.002458733,-1.2277029e-09,-1.799677e+18,1.1118998e-31,-1.6949132e-24,3.645732e+31,-1.1738217e-29,2.5184742e+25,0.0580269,2.3538884e+07,-2.1701745e-16,1.4470146e-36,-2.7275688e-30,7.383211e+16,-1.0973442e-18,-5.2464697e-31,-1.402948e-17,-9.9831006e-20,2.1272136e+17,-2.687164e-29,-1.3445954e-17,-2.2170227e+35,9.29002e+14,5.804833e-32,1.7297585e-23,2.5387039e-24,5.5037823e-09,8.355657e-29,1.8565212e+22,-8.251928e+21,1.4865286e-27,-1.806039e+26,-4.5492703e+14,8.882903e-05,8.831931e-11,-1.8769781e+28,-1.8054307e-27,-8.648073,-1.6928041e-08,-126.841515,-6.7038026e+23,-1.0161431e+23,9.3316386e+23,0.010362554,-3.3322818e+15,6.7174377e+22]':::VECTOR, e'nJ\x10':::STRING, e'A\x10-4':::STRING)
        		AS col_766095,
        	337518288:::REGPROC AS col_766096
        LIMIT
        	80:::INT8;
```
Schema:

```
    rsg_test.go:766: To reproduce, use schema:
    rsg_test.go:768: SET sql_safe_updates = false;;
    rsg_test.go:768: SET CLUSTER SETTING sql.stats.automatic_collection.enabled = false;;
    rsg_test.go:768: SET CLUSTER SETTING sql.stats.histogram_collection.enabled = false;;
    rsg_test.go:768: SET CLUSTER SETTING sql.stats.forecasts.enabled = false;;
    rsg_test.go:768: CREATE TABLE "tabLe_1" ("'col1_0" BYTES NOT NULL, " col1_1" UUID[] NOT NULL, col1_2 VARCHAR NOT NULL, col1_3 REGROLE NULL, "c%v😨ol1_4" STRING NOT NULL, "col1\\x5d_5" BOX2D NOT NULL, col1_6 VECTOR NULL, col1_7 STRING NOT NULL AS (lower(CAST("col1\\x5d_5" AS STRING))) VIRTUAL, col1_8 STRING NOT NULL AS (lower(col1_2)) VIRTUAL, PRIMARY KEY ("c%v😨ol1_4", col1_7 ASC), INDEX ("'col1_0" DESC) NOT VISIBLE, INDEX (" col1_1") NOT VISIBLE, UNIQUE (col1_2 DESC), UNIQUE ("col1\\x5d_5" DESC, col1_7, col1_3, "c%v😨ol1_4", (CASE WHEN col1_3 IS NULL THEN e'm\x13':::STRING ELSE '':::STRING END), lower(CAST(col1_6 AS STRING)), col1_2 ASC, " col1_1", col1_8 DESC), INDEX ("col1\\x5d_5"), INVERTED INDEX (col1_3 ASC, " col1_1"), INDEX (col1_8 DESC) WHERE (((("tabLe_1".col1_2 > '"':::STRING) OR ("tabLe_1".col1_8 < '"':::STRING)) AND ("tabLe_1"."c%v😨ol1_4" <= e'\'':::STRING)) OR ("tabLe_1".col1_7 != '':::STRING)) AND ("tabLe_1"."'col1_0" <= '\xff':::BYTES), INVERTED INDEX (col1_8 gin_trgm_ops), UNIQUE (col1_2), UNIQUE (col1_2, col1_3));
    rsg_test.go:768: CREATE TABLE table_2 (" col2_0" TIMESTAMP NULL, col2_1 FLOAT8 NOT NULL, col2_2 REGTYPE NULL, col2_3 REFCURSOR NOT NULL, "c""ol2{_4" INT2, coͤl😆2_5 TSVECTOR NOT NULL, col2_6 BPCHAR NOT NULL, col2_7 INT8, "%qcol2_8" STRING COLLATE da_DK NOT NULL, col😖2_9 STRING COLLATE en_US NULL, "coL2_10" BIT(4) NOT NULL, col2_11 DATE NOT NULL, col2_12 INTERVAL NOT NULL, "co\\x60l😽2_13" JSONB NOT NULL, col2_14 PG_LSN, "c
ol2_15" NAME NOT NULL, col2_16 OID, c̕ol2_17 TSVECTOR, INDEX ("coL2_10"), UNIQUE (col2_6));
    rsg_test.go:768: CREATE TABLE "tablE_3" ("c o'l%p3_0" REGCLASS, "col\\u484F3_1" JSONB NOT NULL, "co l3_2" STRING COLLATE en_US NOT NULL, col3_3 INT4 NOT NULL, "col%p3 _4" PG_LSN NULL, col3_5 BPCHAR NOT NULL, col3_6 REGROLE NULL, "Col3_7" VARCHAR NOT NULL, col3_8 STRING NOT NULL AS (lower(col3_5)) STORED, "c ol3_9" STRING AS (CASE WHEN "c o'l%p3_0" IS NULL THEN e'Y\nKut':::STRING ELSE 'igX':::STRING END) STORED, "co l%p3_10" STRING NULL AS (lower(CAST("col%p3 _4" AS STRING))) VIRTUAL, "
col3_11" STRING AS (CASE WHEN "c o'l%p3_0" IS NULL THEN 'u':::STRING ELSE '#>b<':::STRING END) VIRTUAL, "col""3_12" STRING NOT NULL AS (lower(col3_5)) VIRTUAL, PRIMARY KEY (col3_8 DESC), INDEX (col3_3), INDEX ("col\\u484F3_1" ASC, col3_3 DESC, "c o'l%p3_0" ASC, col3_6 ASC, "col""3_12" ASC, "col%p3 _4" DESC, col3_8, col3_5 ASC, "Col3_7" ASC, "co l3_2" DESC, "co l%p3_10" DESC, lower("Col3_7"), "c ol3_9" ASC) WHERE (("tablE_3"."co l%p3_10" = '':::STRING) OR ("tablE_3"."
col3_11" <= e'\U00002603':::STRING)) AND ("tablE_3"."c ol3_9" > e'\U00002603':::STRING), INDEX ("co l3_2" DESC) STORING ("c o'l%p3_0", col3_3, "Col3_7", "c ol3_9"), INDEX ("c ol3_9" DESC, (CASE WHEN "c o'l%p3_0" IS NULL THEN '8':::STRING ELSE '':::STRING END) DESC, col3_3 DESC));
    rsg_test.go:768: CREATE TABLE " table_4" (col😭4_0 INET, "col 4_1" OID NULL, col4_2 OID, "col 4_3" DECIMAL NOT NULL, col4_4 GEOMETRY NOT NULL, "col 4_5" INET NOT NULL, col4_6 VECTOR NOT NULL, col4_7 TSVECTOR NOT NULL, "c\\xb0Ol4_8" INTERVAL NOT NULL, col4_9 REGTYPE, col4_10 BYTES NULL, c̸o̴l4_11 INTERVAL NOT NULL, col4_12 INT8 NULL, col4_13 BOX2D NOT NULL, "c'ol4_14" BIT(25) NOT NULL, "col4%v_15" STRING NOT NULL AS (lower(CAST(col4_4 AS STRING))) STORED, col4_16 STRING NOT NULL AS (lower(CAST(col4_7 AS STRING))) STORED, PRIMARY KEY (c̸o̴l4_11), UNIQUE (col4_12 DESC, "col 4_1" ASC) STORING (col4_4, col4_6, col4_7, "c\\xb0Ol4_8", col4_9, "c'ol4_14", "col4%v_15"), UNIQUE (col4_16 DESC, col😭4_0 DESC, col4_13) WHERE ((((" table_4".col4_12 = 2147483647:::INT8) OR (" table_4".col4_16 > e'\x00':::STRING)) OR (" table_4"."col4%v_15" <= e'\x00':::STRING)) AND (" table_4"."col 4_3" = 'NaN':::DECIMAL)) AND (" table_4".col4_10 != '\x22':::BYTES), UNIQUE (col4_10));
    rsg_test.go:768: CREATE TABLE ta͚ble_5 (col5_0 NAME NOT NULL, col5_1 "char" NOT NULL, col5_2 INT8 NOT NULL, col5_3 STRING COLLATE da_DK NOT NULL, " co\\x70l-5_4" REGCLASS, "cOl5_5" TIMESTAMP NOT NULL, col5😥_6 INT8 NOT NULL AS (abs(col5_2)) STORED, INDEX (" co\\x70l-5_4", col5_3 ASC) WHERE (((ta͚ble_5.col5_0 > e'\x00':::STRING:::NAME) AND (ta͚ble_5.col5_2 = 9223372036854775807:::INT8)) OR (ta͚ble_5."cOl5_5" = '2001-01-01 00:00:00 BC':::TIMESTAMP)) OR (ta͚ble_5.col5😥_6 <= (-32768):::INT8), UNIQUE (" co\\x70l-5_4") PARTITION BY LIST (" co\\x70l-5_4") (PARTITION ta͚ble_5_part_0 VALUES IN ((3323191455:::REGCLASS,), (3309906878:::REGCLASS,), (0:::OID,)), PARTITION ta͚blͅe_5_part_1 VALUES IN ((1306046787:::REGCLASS,), (146494704:::REGCLASS,), (234110070:::REGCLASS,), (1892326436:::REGCLASS,)), PARTITION ta͚ble_5_part_2 VALUES IN ((3459672574:::REGCLASS,), (1844064458:::REGCLASS,), (4269923982:::REGCLASS,), (3633795871:::REGCLASS,)), PARTITION "ta{ 😅͚ble_5_part_3" VALUES IN ((3919249822:::REGCLASS,), (530295914:::REGCLASS,), (844341151:::REGCLASS,), (1684863971:::REGCLASS,)), PARTITION "ta͚!ble_5_part_4" VALUES IN ((1766218677:::REGCLASS,), (3327767117:::REGCLASS,), (1665123011:::REGCLASS,), (4147313782:::REGCLASS,)), PARTITION ta͚ble_5_part_5 VALUES IN ((1617095342:::REGCLASS,), (1010318653:::REGCLASS,), (1047253737:::REGCLASS,)), PARTITION ta͚ble_5_part_6 VALUES IN ((1477783625:::REGCLASS,), (3091026932:::REGCLASS,)), PARTITION "DEFAULT" VALUES IN ((DEFAULT,))) WHERE (ta͚ble_5."cOl5_5" > '294276-12-31 23:59:59.999999':::TIMESTAMP) AND (ta͚ble_5.col5_1 < '"':::STRING), UNIQUE (" co\\x70l-5_4" DESC) STORING (col5_2) WHERE (ta͚ble_5.col5_1 >= 'X':::STRING) OR (ta͚ble_5.col5😥_6 != (-9223372036854775807):::INT8), INDEX (" co\\x70l-5_4" ASC, "cOl5_5", col5_3, col5_2 DESC, col5_1 DESC, col5_0) WHERE ((ta͚ble_5."cOl5_5" != '294276-12-31 23:59:59.999999':::TIMESTAMP) OR (ta͚ble_5.col5_2 >= (-32768):::INT8)) OR (ta͚ble_5.col5_0 >= '':::STRING:::NAME), UNIQUE (" co\\x70l-5_4", lower(col5_0), col5😥_6 DESC) STORING (col5_1, col5_2, col5_3, "cOl5_5") PARTITION BY LIST (" co\\x70l-5_4") (PARTITION "ta͚ble_5_pa'rt_0" VALUES IN ((1629366078:::REGCLASS,), (522283900:::REGCLASS,), (2742356182:::REGCLASS,), (2705624405:::REGCLASS,), (364064142:::REGCLASS,), (0:::OID,), (4045228944:::REGCLASS,), (2431087990:::REGCLASS,), (2357907005:::REGCLASS,), (3981307246:::REGCLASS,)), PARTITION ta͚ble_5_part_1 VALUES IN ((2435359776:::REGCLASS,), (3589160999:::REGCLASS,), (2078087308:::REGCLASS,), (1048255423:::REGCLASS,), (441294695:::REGCLASS,), (4165590533:::REGCLASS,), (4255521396:::REGCLASS,), (2827086733:::REGCLASS,), (1839672167:::REGCLASS,)), PARTITION "ta͚ble
_5_part_2" VALUES IN ((890173587:::REGCLASS,), (2198329895:::REGCLASS,), (4008503547:::REGCLASS,), (2779949611:::REGCLASS,), (1366855550:::REGCLASS,), (1771506607:::REGCLASS,), (3174181732:::REGCLASS,), (639444865:::REGCLASS,)), PARTITION ta͚ble_5_part_3 VALUES IN ((2749106685:::REGCLASS,), (1802963618:::REGCLASS,), (3870453446:::REGCLASS,), (4174915473:::REGCLASS,), (4212270949:::REGCLASS,), (3015644908:::REGCLASS,), (3058091647:::REGCLASS,), (3494880417:::REGCLASS,), (363438696:::REGCLASS,), (44840514:::REGCLASS,)), PARTITION "ta͚bl\\uAEA9e_5_part_4" VALUES IN ((523655724:::REGCLASS,), (2830501896:::REGCLASS,), (273125292:::REGCLASS,), (1916351106:::REGCLASS,), (4186871157:::REGCLASS,), (1287677309:::REGCLASS,), (3223084574:::REGCLASS,), (3512741312:::REGCLASS,), (2431006506:::REGCLASS,), (2322807852:::REGCLASS,)), PARTITION ta͚ble_5_part_5 VALUES IN ((2911103264:::REGCLASS,), (2474997913:::REGCLASS,), (1864694219:::REGCLASS,), (203440322:::REGCLASS,), (3382658400:::REGCLASS,), (445038523:::REGCLASS,), (3090314615:::REGCLASS,), (1481903452:::REGCLASS,), (814543523:::REGCLASS,)), PARTITION "t'a͚ble_5_par\\u51E6t_6" VALUES IN ((177093887:::REGCLASS,), (875914779:::REGCLASS,), (1259304971:::REGCLASS,), (4200937309:::REGCLASS,), (3172210961:::REGCLASS,), (1860934576:::REGCLASS,), (3502571002:::REGCLASS,), (3016467395:::REGCLASS,), (2848992227:::REGCLASS,), (175072032:::REGCLASS,))) WHERE (((ta͚ble_5.col5😥_6 != 32767:::INT8) OR (ta͚ble_5.col5_2 != (-9223372036854775807):::INT8)) OR (ta͚ble_5.col5_0 < e'\U00002603':::STRING:::NAME)) AND (ta͚ble_5."cOl5_5" != '0001-01-01 00:00:00':::TIMESTAMP), INDEX (col5_1 ASC) STORING (col5_0, " co\\x70l-5_4") PARTITION BY LIST (col5_1) (PARTITION ta͚ble_5_part_0 VALUES IN (('u':::STRING,), ('8':::STRING,), ('G':::STRING,)), PARTITION ta͚ble_5_part_1 VALUES IN (('Y':::STRING,), ('Z':::STRING,), (e'\x00':::STRING,)), PARTITION "t a͚ble_5_part_2" VALUES IN (('_':::STRING,), ('b':::STRING,), ('X':::STRING,)), PARTITION "ta͚bl%qe_5_part_3" VALUES IN (('|':::STRING,), ('-':::STRING,)), PARTITION "ta͚b%qle_5_part_4" VALUES IN (('k':::STRING,), ('6':::STRING,), (e'\U00002603':::STRING,)), PARTITION ta͚ble_5_part_5 VALUES IN ((e'\x16':::STRING,), ('9':::STRING,), ('&':::STRING,))) WHERE ((((ta͚ble_5.col5_1 != e'\x00':::STRING) OR (ta͚ble_5.col5😥_6 < (-1):::INT8)) AND (ta͚ble_5.col5_0 > e'\'':::STRING:::NAME)) OR (ta͚ble_5.col5_2 >= (-1):::INT8)) AND (ta͚ble_5."cOl5_5" != '294276-12-31 23:59:59.999999':::TIMESTAMP) NOT VISIBLE, UNIQUE (lower(col5_0) DESC), INDEX (col5_0 DESC) WHERE ((((ta͚ble_5.col5_0 = e'\U00002603':::STRING:::NAME) AND (ta͚ble_5.col5_2 > 2147483647:::INT8)) AND (ta͚ble_5.col5_1 = e'\'':::STRING)) AND (ta͚ble_5.col5😥_6 = 1:::INT8)) OR (ta͚ble_5."cOl5_5" < '294276-12-31 23:59:59.999999':::TIMESTAMP), UNIQUE (col5_2 DESC) STORING (col5_0, " co\\x70l-5_4", "cOl5_5", col5😥_6) WHERE ((((ta͚ble_5.col5_2 != 0:::INT8) AND (ta͚ble_5.col5_0 <= '':::STRING:::NAME)) OR (ta͚ble_5."cOl5_5" <= '2001-01-01 00:00:00 BC':::TIMESTAMP)) OR (ta͚ble_5.col5_1 = '':::STRING)) AND (ta͚ble_5.col5😥_6 = (-32768):::INT8), FAMILY (col5_0, col5😥_6), FAMILY (col5_2), FAMILY (col5_3), FAMILY (" co\\x70l-5_4"), FAMILY ("cOl5_5"), FAMILY (col5_1));
    rsg_test.go:768: ALTER TABLE "tabLe_1" INJECT STATISTICS e'[{"avg_size": 18, "columns": ["\\"\'col1_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3688519887682886058, "histo_buckets": [{"distinct_range": 0, "num_eq": 6638447145435667418, "num_range": 0, "upper_bound": "\\\\xa86f6d7fe90f70f6"}], "histo_col_type": "BYTES", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7713812913447765679}, {"avg_size": 5, "columns": ["\\" col1_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6029893973933673920, "histo_buckets": [{"distinct_range": 0, "num_eq": 70000000000, "num_range": 0, "upper_bound": "ARRAY[\'e72f4ec6-86ae-4711-9efe-180519d11f2a\':::UUID]"}, {"distinct_range": 688515436532341100, "num_eq": 6225988439084449970, "num_range": 1410747066310101557, "upper_bound": "ARRAY[\'b0cf4b7b-5fc6-4a0c-8743-8abb96ec29d9\':::UUID,\'2ba6fdef-b832-4bd6-a934-06eb99ecdfd7\':::UUID]"}, {"distinct_range": 0, "num_eq": 80000000000, "num_range": 1000, "upper_bound": "ARRAY[\'e13cd8c8-e789-4a0a-a813-af060e14bd1b\':::UUID,\'ba289454-28f1-4851-a694-bc13753934e0\':::UUID,\'54121a6e-3038-4e17-8511-a448743edfa9\':::UUID,\'284887e3-4d23-4564-867b-fd3f9ac0f39c\':::UUID,\'5b70ea7e-12f3-43e1-9001-b14e57f47ffd\':::UUID]"}, {"distinct_range": 594897964201007500, "num_eq": 7000000000, "num_range": 1195212071403568005, "upper_bound": "ARRAY[\'ffffffff-ffff-ffff-ffff-ffffffffffff\':::UUID,\'16a3f53d-128f-4689-bafb-6d33757ebdb3\':::UUID,\'4fc8f4a4-5e66-449d-bbf2-e0dce734c236\':::UUID,\'b5b4e446-e47d-4264-a9b8-2e93bdaa15c4\':::UUID,\'0f80ff7b-27c5-40ab-bbc6-3bdcd180021d\':::UUID]"}, {"distinct_range": 0, "num_eq": 400, "num_range": 547550575100516378, "upper_bound": "ARRAY[\'b1857513-d9cc-48a3-80e1-e84c4be1ee03\':::UUID,\'22448760-492b-410b-99f3-7b4d90d2d3f4\':::UUID,\'02152041-8950-4e16-931a-bac32ca7b4d9\':::UUID,\'df9f9887-5325-46e1-870b-61e7dbf03bc2\':::UUID,\'28e08238-3d81-4718-b93e-da93721ddcca\':::UUID,\'3565e36f-0f9e-4226-aea9-c9d27dabfde5\':::UUID]"}], "histo_col_type": "UUID[]", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7713812913447765679}, {"avg_size": 21, "columns": ["col1_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2268530356170886127, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 7713812913447765679}, {"avg_size": 31, "columns": ["col1_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5872474547452480204, "histo_col_type": "", "name": "__auto__", "null_count": 3856408807168397233, "row_count": 7713812913447765679}, {"avg_size": 25, "columns": ["\\"c%v\U0001F628ol1_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 5879944791550348829, "histo_buckets": [{"distinct_range": 0, "num_eq": 400, "num_range": 0, "upper_bound": "$,"}, {"distinct_range": 0, "num_eq": 9109903382867844129, "num_range": 6000000, "upper_bound": "pN"}, {"distinct_range": 0, "num_eq": 4013685873388629585, "num_range": 5968835250713363136, "upper_bound": "\'z|"}, {"distinct_range": 0, "num_eq": 20000000, "num_range": 677751180451814812, "upper_bound": "\\u0010}M\\u0015%"}, {"distinct_range": 0, "num_eq": 5000000000, "num_range": 500000000, "upper_bound": "NF\\u0016\\u0001f$\\u0005"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7713812913447765679}, {"avg_size": 13, "columns": ["\\"col1\\\\\\\\x5d_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4492783641438145816, "histo_buckets": [{"distinct_range": 0, "num_eq": 6701300073293671819, "num_range": 0, "upper_bound": "BOX(-10 -10,10 10)"}, {"distinct_range": 48862659.167922094, "num_eq": 9000, "num_range": 60000000, "upper_bound": "BOX(-0.6186485249775244 -0.07736105565237072,-0.26859666867082854 0.41431807120944786)"}, {"distinct_range": 5714701.483173107, "num_eq": 1711671799762633459, "num_range": 20000000, "upper_bound": "BOX(-0.6040781756454785 -0.854907403546487,-0.1193679823866238 0.24595214686728756)"}, {"distinct_range": 22830323.55480514, "num_eq": 3755264854953035486, "num_range": 800000000, "upper_bound": "BOX(-0.3057557804314931 -0.10534081615145319,0.025526757137172618 1.4094913318960887)"}, {"distinct_range": 1122921050867390700, "num_eq": 700, "num_range": 2037639731633637224, "upper_bound": "BOX(-0.1304783518934732 -2.439023770333762,0.8239122285867451 -0.4333982121542988)"}, {"distinct_range": 483661191.170449, "num_eq": 6020864877271941544, "num_range": 600000000, "upper_bound": "BOX(0.27041447590828194 -0.7384713985330028,1.501000156921096 -0.7378625620222156)"}, {"distinct_range": 0, "num_eq": 4683803124105904585, "num_range": 70000, "upper_bound": "BOX(1.0319216599243912 -1.3997672391467053,1.3225123611603744 0.11983600170441844)"}], "histo_col_type": "BOX2D", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7713812913447765679}, {"avg_size": 11, "columns": ["col1_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 6071326497530113241, "histo_col_type": "", "name": "__auto__", "null_count": 7201981187432911502, "row_count": 7713812913447765679}, {"avg_size": 8, "columns": ["col1_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 593223218881085795, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 7713812913447765679}, {"avg_size": 22, "columns": ["col1_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 862516507851594304, "histo_buckets": [{"distinct_range": 0, "num_eq": 5167436227489596669, "num_range": 0, "upper_bound": "$"}, {"distinct_range": 7707631210563662000, "num_eq": 80000, "num_range": 8510948291042267445, "upper_bound": "yI"}, {"distinct_range": 8846382005870733000, "num_eq": 20, "num_range": 8846382005870733792, "upper_bound": "p9o\\u001f@!U\\u000cw"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 7713812913447765679}]':::JSONB;
    rsg_test.go:768: ALTER TABLE table_2 INJECT STATISTICS e'[{"avg_size": 7, "columns": ["\\" col2_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3822322287673400271, "histo_col_type": "", "name": "__auto__", "null_count": 2323935854466382426, "row_count": 4548557781106546364}, {"avg_size": 15, "columns": ["col2_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2811664834759846119, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 29, "columns": ["col2_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4171954953080215884, "histo_col_type": "", "name": "__auto__", "null_count": 3273458119822157316, "row_count": 4548557781106546364}, {"avg_size": 26, "columns": ["col2_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 851226491305671891, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 24, "columns": ["\\"c\\"\\"ol2{_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 375724393397473303, "histo_col_type": "", "name": "__auto__", "null_count": 3126753535714532981, "row_count": 4548557781106546364}, {"avg_size": 4, "columns": ["co\u0364l\U0001F6062_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3513267884898420663, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 0, "columns": ["col2_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1651756535658071670, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 17, "columns": ["col2_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2710032895478217463, "histo_col_type": "", "name": "__auto__", "null_count": 3697291371260868542, "row_count": 4548557781106546364}, {"avg_size": 23, "columns": ["\\"%qcol2_8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3862083902089404142, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 5, "columns": ["col\U0001F6162_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3570628979120390525, "histo_col_type": "", "name": "__auto__", "null_count": 4547746700001379912, "row_count": 4548557781106546364}, {"avg_size": 17, "columns": ["\\"coL2_10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4228146680795880563, "histo_buckets": [{"distinct_range": 0, "num_eq": 80000000, "num_range": 0, "upper_bound": "0000"}, {"distinct_range": 311797856057502200, "num_eq": 8000000, "num_range": 741988237701438398, "upper_bound": "0010"}, {"distinct_range": 3650592.976643315, "num_eq": 30000000000, "num_range": 7000000, "upper_bound": "1001"}], "histo_col_type": "BIT(4)", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 26, "columns": ["col2_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2339924101157328046, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 30, "columns": ["col2_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 429996153796326322, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 0, "columns": ["\\"co\\\\\\\\x60l\U0001F63D2_13\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1139197510694247384, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 22, "columns": ["col2_14"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3491624149653111275, "histo_col_type": "", "name": "__auto__", "null_count": 2748990831940728756, "row_count": 4548557781106546364}, {"avg_size": 9, "columns": ["\\"c\\rol2_15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4259043924406625399, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 4548557781106546364}, {"avg_size": 25, "columns": ["col2_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 4100945832957968997, "histo_col_type": "", "name": "__auto__", "null_count": 374512797496690768, "row_count": 4548557781106546364}, {"avg_size": 0, "columns": ["c\u0315ol2_17"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 3325149981710491005, "histo_col_type": "", "name": "__auto__", "null_count": 2718875294211967815, "row_count": 4548557781106546364}]':::JSONB;
    rsg_test.go:768: ALTER TABLE "tablE_3" INJECT STATISTICS e'[{"avg_size": 1, "columns": ["\\"c o\'l%p3_0\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 641484215740979657, "histo_col_type": "", "name": "__auto__", "null_count": 761052555493994970, "row_count": 1127829937180338929}, {"avg_size": 28, "columns": ["\\"col\\\\\\\\u484F3_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 63808929519739560, "histo_buckets": [{"distinct_range": 0, "num_eq": 40000, "num_range": 0, "upper_bound": "null"}, {"distinct_range": 609821486787383900, "num_eq": 7605524062430078915, "num_range": 3118760841542052810, "upper_bound": "{\\"b\\": false, \\"wlPqBOC0\\": 0.10497984126929469}"}, {"distinct_range": 17722634709.947968, "num_eq": 3000000, "num_range": 30000000000, "upper_bound": "{\\"Vyfjekm\\": null, \\"b\\": [3.7114225858555487, 1.9147360684420458]}"}, {"distinct_range": 2000000000, "num_eq": 8349383494820628363, "num_range": 2000000000, "upper_bound": "{\\"baz\\": true, \\"foobar\\": [\\"c\\", null], \\"n8Ovc6l\\": [0.05235494198684798]}"}, {"distinct_range": 6357775129561497000, "num_eq": 8169637605787053193, "num_range": 6357775129561496252, "upper_bound": "[{\\"WVVzR\\": \\"8XZ5GW0\\"}, {\\"Ubs\\": false, \\"b\\": true, \\"foo\\": {\\"MXB4OuD\\": false}}, 0.3286143838275584, \\"hX1\\"]"}, {"distinct_range": 4197781503107206700, "num_eq": 846381993256792681, "num_range": 5884176044695557719, "upper_bound": "{\\"LO37\\": null, \\"baz\\": \\"foo\\", \\"c\\": 0.9950612765196329, \\"foo\\": {\\"2M\\": {\\"cnoHb1\\": 0.21342159045582526}, \\"baz\\": \\"fO1FV\\", \\"nJQzd\\": false}}"}], "histo_col_type": "JSONB", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 1127829937180338929}, {"avg_size": 15, "columns": ["\\"co l3_2\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 914667767225335339, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1127829937180338929}, {"avg_size": 25, "columns": ["col3_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 478568029082053077, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1127829937180338929}, {"avg_size": 17, "columns": ["\\"col%p3 _4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 386188710731812019, "histo_col_type": "", "name": "__auto__", "null_count": 1084838161141780830, "row_count": 1127829937180338929}, {"avg_size": 13, "columns": ["col3_5"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 549202142244523135, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1127829937180338929}, {"avg_size": 25, "columns": ["col3_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1046545129045719235, "histo_col_type": "", "name": "__auto__", "null_count": 750877334527160863, "row_count": 1127829937180338929}, {"avg_size": 7, "columns": ["\\"Col3_7\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 21810584157173127, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1127829937180338929}, {"avg_size": 5, "columns": ["col3_8"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 562384055248778967, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1127829937180338929}, {"avg_size": 6, "columns": ["\\"c ol3_9\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 341831903767316712, "histo_buckets": [{"distinct_range": 0, "num_eq": 40000000000, "num_range": 0, "upper_bound": ""}, {"distinct_range": 2266167317819556000, "num_eq": 4411089171654452560, "num_range": 7850688346578185284, "upper_bound": "\\""}, {"distinct_range": 0, "num_eq": 7231253206783577597, "num_range": 90000000, "upper_bound": "P4)i\\r\\u0007"}, {"distinct_range": 69930190.41505976, "num_eq": 200, "num_range": 200000000, "upper_bound": "#w\\u00024wM\\r/"}, {"distinct_range": 0, "num_eq": 0, "num_range": 3733219747745504610, "upper_bound": "lc;J\\u0003(fx"}], "histo_col_type": "STRING", "histo_version": 3, "name": "__auto__", "null_count": 132436502276463657, "row_count": 1127829937180338929}, {"avg_size": 24, "columns": ["\\"co l%p3_10\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 247266274035566180, "histo_col_type": "", "name": "__auto__", "null_count": 958879390063730647, "row_count": 1127829937180338929}, {"avg_size": 10, "columns": ["\\"\\rcol3_11\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 997831719012677157, "histo_col_type": "", "name": "__auto__", "null_count": 836787540032901337, "row_count": 1127829937180338929}, {"avg_size": 25, "columns": ["\\"col\\"\\"3_12\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 314331309994310432, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 1127829937180338929}]':::JSONB;
    rsg_test.go:768: ALTER TABLE " table_4" INJECT STATISTICS e'[{"avg_size": 19, "columns": ["col\U0001F62D4_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 167095463585823603, "histo_col_type": "", "name": "__auto__", "null_count": 1204242709787132058, "row_count": 2852796474558310247}, {"avg_size": 21, "columns": ["\\"col 4_1\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1628705388411189782, "histo_col_type": "", "name": "__auto__", "null_count": 2174892832581539588, "row_count": 2852796474558310247}, {"avg_size": 12, "columns": ["col4_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1889340638541633739, "histo_col_type": "", "name": "__auto__", "null_count": 2474242781455727283, "row_count": 2852796474558310247}, {"avg_size": 26, "columns": ["\\"col 4_3\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1511876412868142308, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 26, "columns": ["col4_4"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2330659623926636576, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 17, "columns": ["\\"col 4_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 869984854458047733, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 8, "columns": ["col4_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 1769357414311755605, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 31, "columns": ["col4_7"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 58213325248174885, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 26, "columns": ["\\"c\\\\\\\\xb0Ol4_8\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2344672501027861159, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 21, "columns": ["col4_9"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 170505717722727513, "histo_col_type": "", "name": "__auto__", "null_count": 2591266562649908872, "row_count": 2852796474558310247}, {"avg_size": 26, "columns": ["col4_10"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2143021404489679636, "histo_col_type": "", "name": "__auto__", "null_count": 2841353969744306238, "row_count": 2852796474558310247}, {"avg_size": 26, "columns": ["c\u0338o\u0334l4_11"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 53615163641013566, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 27, "columns": ["col4_12"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 330157643166615606, "histo_buckets": [{"distinct_range": 0, "num_eq": 7484943553786323663, "num_range": 0, "upper_bound": "-5535757544893510920"}, {"distinct_range": 3531895620739550700, "num_eq": 90000, "num_range": 8496100686775545632, "upper_bound": "6247937715742643853"}, {"distinct_range": 0, "num_eq": 10000, "num_range": 100, "upper_bound": "-8241614578462520729"}, {"distinct_range": 2393357435728533500, "num_eq": 8391050304003343124, "num_range": 2393357435728533618, "upper_bound": "-1731198280130754975"}, {"distinct_range": 3274159896112525300, "num_eq": 60000000, "num_range": 3306453126553802965, "upper_bound": "6469367965654228783"}], "histo_col_type": "INT8", "histo_version": 3, "name": "__auto__", "null_count": 1068546552936136211, "row_count": 2852796474558310247}, {"avg_size": 28, "columns": ["col4_13"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 346569033861883521, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 4, "columns": ["\\"c\'ol4_14\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 2002787798681153437, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 23, "columns": ["\\"col4%v_15\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 106940202189082659, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}, {"avg_size": 23, "columns": ["col4_16"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 635467204306163306, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 2852796474558310247}]':::JSONB;
    rsg_test.go:768: ALTER TABLE ta͚ble_5 INJECT STATISTICS e'[{"avg_size": 25, "columns": ["col5_0"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 9971, "histo_buckets": [{"distinct_range": 0, "num_eq": 400000, "num_range": 0, "upper_bound": ""}, {"distinct_range": 0, "num_eq": 864363363454414379, "num_range": 9000000, "upper_bound": "\'"}, {"distinct_range": 2307060641249003500, "num_eq": 7248427602724404302, "num_range": 2307060641249003543, "upper_bound": "%\\u000e\\u0016\\u000cx\\u0013"}, {"distinct_range": 0, "num_eq": 5447265804470191214, "num_range": 2011808755801746250, "upper_bound": "b7uZ4RL"}], "histo_col_type": "NAME", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 400000}, {"avg_size": 16, "columns": ["col5_1"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 123893, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 400000}, {"avg_size": 21, "columns": ["col5_2"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 105895, "histo_buckets": [{"distinct_range": 0, "num_eq": 30000, "num_range": 0, "upper_bound": "7486421831809024489"}, {"distinct_range": 70000, "num_eq": 6642697509284588320, "num_range": 70000, "upper_bound": "-8767810899107210285"}, {"distinct_range": 0, "num_eq": 5067137562420097488, "num_range": 20000, "upper_bound": "-6318829852945058174"}], "histo_col_type": "INT8", "histo_version": 3, "name": "__auto__", "null_count": 0, "row_count": 400000}, {"avg_size": 26, "columns": ["col5_3"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 270746, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 400000}, {"avg_size": 28, "columns": ["\\" co\\\\\\\\x70l-5_4\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 126743, "histo_buckets": [{"distinct_range": 0, "num_eq": 3840501758711442043, "num_range": 0, "upper_bound": "2500595041"}, {"distinct_range": 2463872843940312600, "num_eq": 5526239306047408491, "num_range": 2463872843940312538, "upper_bound": "346788785"}], "histo_col_type": "REGCLASS", "histo_version": 3, "name": "__auto__", "null_count": 303774, "row_count": 400000}, {"avg_size": 25, "columns": ["\\"cOl5_5\\""], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 87252, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 400000}, {"avg_size": 21, "columns": ["col5\U0001F625_6"], "created_at": "2000-01-01 00:00:00+00:00", "distinct_count": 282044, "histo_col_type": "", "name": "__auto__", "null_count": 0, "row_count": 400000}]':::JSONB;
    rsg_test.go:768: ALTER TABLE "tabLe_1" ADD FOREIGN KEY (col1_2) REFERENCES table_2 (col2_6);
    rsg_test.go:768: ALTER TABLE "tablE_3" ADD FOREIGN KEY ("Col3_7") REFERENCES "tabLe_1" (col1_2) ON DELETE CASCADE ON UPDATE RESTRICT;
    rsg_test.go:768: ALTER TABLE "tabLe_1" ADD FOREIGN KEY ("'col1_0") REFERENCES " table_4" (col4_10) ON DELETE RESTRICT;
    rsg_test.go:768: ALTER TABLE "tablE_3" ADD FOREIGN KEY (col3_5, "c o'l%p3_0") REFERENCES "tabLe_1" (col1_2, col1_3);
    rsg_test.go:768: CREATE TYPE rand_typ_0 AS (gsyu TIMESTAMPTZ, hn INT8, iovau INET, t DECIMAL, n BOX2D, i TIMESTAMPTZ);
    rsg_test.go:768: SET sql_safe_updates = false;;
    rsg_test.go:768: 
        BEGIN; CREATE TYPE greeting AS ENUM ('hello', 'howdy', 'hi', 'good day', 'morning'); COMMIT;
        BEGIN;
        CREATE TABLE IF NOT EXISTS seed AS
        	SELECT
        		g::INT2 AS _int2,
        		g::INT4 AS _int4,
        		g::INT8 AS _int8,
        		g::FLOAT4 AS _float4,
        		g::FLOAT8 AS _float8,
        		'2001-01-01'::DATE + g AS _date,
        		'2001-01-01'::TIMESTAMP + g * '1 day'::INTERVAL AS _timestamp,
        		'2001-01-01'::TIMESTAMPTZ + g * '1 day'::INTERVAL AS _timestamptz,
        		g * '1 day'::INTERVAL AS _interval,
        		g % 2 = 1 AS _bool,
        		g::DECIMAL AS _decimal,
        		g::STRING AS _string,
        		g::STRING::BYTES AS _bytes,
        		substring('00000000-0000-0000-0000-' || g::STRING || '00000000000', 1, 36)::UUID AS _uuid,
        		'0.0.0.0'::INET + g AS _inet,
        		g::STRING::JSONB AS _jsonb,
        		enum_range('hello'::greeting)[g] as _enum
        	FROM
        		generate_series(1, 5) AS g;
        COMMIT;
        
        INSERT INTO seed DEFAULT VALUES;
        CREATE INDEX on seed (_int8, _float8, _date);
        CREATE INVERTED INDEX on seed (_jsonb);
        ;
    rsg_test.go:768: ALTER TABLE seed SPLIT AT VALUES (1376000904149020474);
    rsg_test.go:768: ALTER TABLE seed SPLIT AT VALUES (8925316233860547418);
    rsg_test.go:768: ALTER TABLE seed SPLIT AT VALUES (4158300613343078254);
    rsg_test.go:768: ALTER TABLE seed SCATTER;
    rsg_test.go:770: 
    rsg_test.go:771: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/d437d2c847dfedbc4972f231c3331c8e/logTestRandomSyntaxSQLSmith390191020
--- FAIL: TestRandomSyntaxSQLSmith (303.58s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxSQLSmith.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-51090

--- a/pkg/sql/logictest/testdata/logic_test/builtin_function
+++ b/pkg/sql/logictest/testdata/logic_test/builtin_function
@@ -4694,4 +4694,20 @@ SELECT substring_index('a..b..c..d', '..', -2);
 ----
 c..d
 
+# Verify boundary cases of count input. Found in issue #147516.
+query T
+SELECT substring_index('a..b..c..d', '..', -9223372036854775808);
+----
+a..b..c..d
+
+query T
+SELECT substring_index('a..b..c..d', '..', -9223372036854775807);
+----
+a..b..c..d
+
+query T
+SELECT substring_index('a..b..c..d', '..', 9223372036854775807);
+----
+a..b..c..d
+
 subtest end

--- a/pkg/sql/sem/builtins/builtins.go
+++ b/pkg/sql/sem/builtins/builtins.go
@@ -422,11 +422,11 @@ var regularBuiltins = map[string]builtinDefinition{
 				}
 
 				// If count is negative, return the last 'abs(count)' parts joined by delim
-				count = -count
-				if count >= length {
+				if -count >= length || count == math.MinInt {
 					return tree.NewDString(input), nil // If count exceeds occurrences, return the full string
 				}
-				return tree.NewDString(strings.Join(parts[length-count:], delim)), nil
+				start := length + count // count is negative
+				return tree.NewDString(strings.Join(parts[start:], delim)), nil
 			},
 			Info: "Returns a substring of `input` before `count` occurrences of `delim`.\n" +
 				"If `count` is positive, the leftmost part is returned. If `count` is negative, the rightmost part is returned.",

